app/models/offer.rb

```diff
@@ -1,4 +1,12 @@
 class Offer < ApplicationRecord
   belongs_to :application_choice
   has_many :conditions, class_name: 'OfferCondition'
+
+  def mark_all_conditions_as_met!
+    conditions.each { |condition| condition.update!(status: :met) }
```

Comment:
can we avoid looping over each condition and update in bulk?

Reply:
saw previous comment nvm

Reply:
We have 3 options here:
1. `conditions.update`
2. `conditions.update_all`
3. `conditions.each update!`

I originally didn't go with `update_all` because it skips callbacks and we had auditing. Now we've removed auditing we could go back to that.

The reason we went with this is to get it to throw if saving fails

Reply:
I think if we go with `update` it will ensure callbacks and bulk, but not failing on validations. However will we ever fail on validations here, I think probably unlikely at this stage, what do you think?

Reply:
Since we don't have validations, then I don't think we can fail at the moment. I'd be happy with just `update` here (which will fail if you give it something that's not in the enum)

Reply:
Can we amend the method name then to exclude the bang suffix